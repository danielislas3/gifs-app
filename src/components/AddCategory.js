import React from "react";
import { useState } from "react";
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {


  const [inputValue, setInputValue] = useState("Hola mundo");
  
  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = e => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((cat) => [...cat, inputValue]);
      setInputValue('')
    }

    console.log(e);
  };
  
  return (
    <form onSubmit={handleSubmit}>
      <h2>Add Category</h2>
      <input
        name="cat"
        type="text"
        value={inputValue}
        onChange={handleInputChange}
      />
    </form>
  );
};

AddCategory.propTypes={
  setCategories:PropTypes.func.isRequired
}