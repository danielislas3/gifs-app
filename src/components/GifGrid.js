import React from "react";
import PropTypes from 'prop-types';
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const { data:gifs, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loading && <p>Loading...</p>}
      <div className="card-grid animate__animated animate__fadeIn">
        {gifs.map(gif => (
          <GifGridItem {...gif} key={gif.id} />
        ))}
      </div>
    </>
  );
};
GifGrid.propTypes={
  category:PropTypes.string.isRequired
}
