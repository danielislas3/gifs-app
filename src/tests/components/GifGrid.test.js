import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid/>', () => {
  const category = 'Pepe';

  test('Debe mostrar el copmponenete ', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar items cuando se cargan imagenes', () => {
    const gifs = [
      {
        id: 'abc',
        url: 'https://giphy.com/embed/HdVtsppIpBigM',
        title: 'gift1',
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
