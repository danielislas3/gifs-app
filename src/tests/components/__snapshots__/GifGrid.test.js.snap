// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> Debe mostrar el copmponenete  1`] = `
<Fragment>
  <h3>
    Pepe
  </h3>
  <p>
    Loading...
  </p>
  <div
    className="card-grid animate__animated animate__fadeIn"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid/> debe de mostrar items cuando se cargan imagenes 1`] = `
<Fragment>
  <h3>
    Pepe
  </h3>
  <div
    className="card-grid animate__animated animate__fadeIn"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="gift1"
      url="https://giphy.com/embed/HdVtsppIpBigM"
    />
  </div>
</Fragment>
`;
